### DISCUSSION TITLE
Installationsberscheibung im Wiki

### DISCUSSION THREAD

**Discussion by wiggal** [wiggal/GEN24_Ladesteuerung#33] [General]
Hallo,

ich habe im Wiki mal eine Installationsbeschreibung für den RaspberryPi geschrieben.
Die Beschreibung erfolgte Größtenteils aus dem Gedächtnis, und kann deshalb unvollständig und fehlerhaft sein.

Bitte hier Fehle und Ergänzungen melden.

Danke Wiggal

**Comment by modezz** [comment]
Hallo,

erstmal danke für deine Arbeit - verrichtet seinen Dienst bei mir perfekt!

Um am RPI lauffähig zu sein musste ich noch folgendes nachinstallieren:
/sudo apt install libatlas3-base libgfortran5

**Reply by wiggal** [reply]
Hi, ich hab die Pakete nicht, und es läuft trotzdem.
Mal sehen, ob das noch jemand feststellt.

**Reply by modezz** [reply]
hello, also bei mir zickte numpy rum:
"ImportError: libcblas.so.3: cannot open shared object file: No such file or directory"

Also mal gesucht nach --> proper numpy istallation:
https://www.piwheels.org/project/numpy/

und das fehlen der libcblas.so.3 gefixt mit:
sudo apt install libatlas3-base libgfortran5

**Reply by wiggal** [reply]
Ok, ich kann ja bei gelegenheit so Fehler und abhife ins Wiki übernehemen

Danke dir

**Comment by drdoener** [comment]
Hallo wiggal,

auch von mir schonmal Danke für Deine Arbeit !
Ich versuchte gerade die Installtion auf einem ubuntu-System.
Bei Schritt 3 kam es zu einem Fehler:
sudo apt install python3      -> OK
sudo apt install python3-pip   -> OK
**sudo pip install pyModbusTCP==v0.1.10**   

"error: externally-managed-environment

Ã This environment is externally managed
â°â> To install Python packages system-wide, try apt install
    python3-xyz, where xyz is the package you are trying to
    install.

    If you wish to install a non-Debian-packaged Python package,
    create a virtual environment using python3 -m venv path/to/venv.
    Then use path/to/venv/bin/python and path/to/venv/bin/pip. Make
    sure you have python3-full installed.

    If you wish to install a non-Debian packaged Python application,
    it may be easiest to use pipx install xyz, which will manage a
    virtual environment for you. Make sure you have pipx installed.

    See /usr/share/doc/python3.11/README.venv for more information.

note: If you believe this is a mistake, please contact your Python installation or OS distribution provider. You can override this, at the risk of breaking your Python installation or OS, by passing --break-system-packages.
hint: See PEP 668 for the detailed specification.

Wie geht man hiermit am Besten um?

Danke und VG
Michael

**Reply by wiggal** [reply]
Hab so einen Fehle auch ncoh nicht gesehen.
Hab mal im Internet nach dem Fehle gesucht:

https://stackoverflow.com/questions/75616371/sudo-pip-install-cryptography-error-externally-managed-environment
https://stackoverflow.com/questions/75602063/pip-install-r-requirements-txt-is-failing-this-environment-is-externally-mana

am häufigsten liest folgendes:
pip install xyz --break-system-packages

Versuch es mal, aber ohne Garantie
